### PR TITLE
Make `shell/platform/android` IDE-friendly, and add documentation

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/141043.

Nothing in this change should impact:

- End-user apps
- Testing and/or CI

I've added what I believe is the minimum required to get a reasonable Android Studio experience with the `shell/platform/android` folder. More could definitely be done, but this unblocks me (i.e. https://github.com/flutter/flutter/issues/139702) and adds instructions for the next person.

_I'm open to suggestions on how to improve this further, but unless they are critical, I'd rather land this as-is and review your PRs tweaking my documentation further (I have limited time to work exclusively on docs)._

---

Example after this:

image

### CODE REVIEW

**Comment by matanlurey** [comment]
Only failure is the licensing bot, which I'll fix up before submitting.

File: shell/platform/android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+repositories {
+    google()
+    mavenCentral()
+}
+
+apply plugin: "com.android.library"
+
+android {
+    compileSdk 31
```
Comment (reidbaker) [inline shell/platform/android/build.gradle:23]: This should really be 34. If you can make that change and it does not break please do so. If it breaks just let me know here and merge as is.

File: shell/platform/android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    compileOptions {
```
Comment (reidbaker) [inline shell/platform/android/build.gradle:25]: I think you should remove this for durability.  There is already a gradle warning that setting this value is going to be deprecated.

**Review by reidbaker — APPROVED** [review]

File: shell/platform/android/build.gradle (hunk 1 above)
Comment (matanlurey) [reply shell/platform/android/build.gradle]: Should work fine! Done.

File: shell/platform/android/build.gradle (hunk 2 above)
Comment (matanlurey) [reply shell/platform/android/build.gradle]: Sg. Done!

File: shell/platform/android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            java.srcDirs './'
+            java.include './io.flutter/**'
+        }
+    }
+}
+
+dependencies {
+    implementation 'androidx.annotation:annotation-jvm:1.7.1'
+}
+
```
Comment (jason-simmons) [inline shell/platform/android/build.gradle:37]: Remove the empty line here

**Review by jason-simmons — COMMENTED** [review]
LGTM

File: shell/platform/android/build.gradle [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        main {
+            manifest.srcFile 'AndroidManifest.xml'
+            java.srcDirs './'
+            java.include './io.flutter/**'
+        }
+    }
+}
+
+dependencies {
+    implementation 'androidx.annotation:annotation-jvm:1.7.1'
```
Comment (gmackall) [inline shell/platform/android/build.gradle:35]: Late to the party here, but did you add this to resolve `androidx.annotation...` imports? I tried following the README from this PR, and it works for all imports except the `androidx` ones. Also, this version should probably match the one in [`files.json`](https://github.com/flutter/engine/blob/ee297c08638faedee0dbd649b4913bf41c63410e/tools/androidx/files.json#L38), right?

**Pull request merged**